### ISSUE TITLE
"Warning: useLayoutEffect does nothing on the server" on useSuspenseQuery

### ISSUE THREAD

**Issue by antonybudianto** [apollographql/apollo-client#10361]
Hi, I just tried the new alpha `@apollo/client@alpha` to try new useSuspenseQuery. 
I've setup the sandbox with React 18.2.0 renderToPipeableStream SSR, and everything looks good so far
- Component with useSuspenseQuery only render once on server
- Hydration works (no extra gql call)

Except one, there is warning about useLayoutEffect every render.

**Intended outcome:**

No warnings when using useSuspenseQuery

**Actual outcome:**

Warning occurs for each SSR page hit:

```
Warning: useLayoutEffect does nothing on the server, because
its effect cannot be encoded into the server renderer's output format. 
This will lead to a mismatch between the initial, non-hydrated UI and the intended UI.
To avoid this, useLayoutEffect should only be used in components 
that render exclusively on the client. See https://reactjs.org/link/uselayouteffect-ssr for common fixes.
```

**How to reproduce the issue:**

1. Open https://codesandbox.io/s/playground-react18-ssr-apollo-suspense-pg4ebk
2. See the warnings on the console

**Versions**

`@apollo/client@alpha`

**Comment by jerelmiller** [comment]
@antonybudianto thanks so much for trying this and reporting this issue! I'll make sure the hook adapts properly when used on the server. I'm glad its working as well as it is already!

**Comment by antonybudianto** [comment]
Sure! Glad I can be of any help, also thanks for the hard work in supporting Suspense 👍🏻

**Comment by alessbell** [comment]
Thanks again for the report! This is fixed in `v3.8.0-alpha.3` 🎉

**Comment by antonybudianto** [comment]
Thanks a lot @alessbell! 👍🏻 🎉
